collect_app/src/main/java/org/odk/collect/android/utilities/ResetUtility.java

```diff
@@ -71,14 +72,29 @@
     }
 
     private void resetPreferences(Context context) {
-        PreferenceManager
+        boolean clearedDefaultPreferences = PreferenceManager
                 .getDefaultSharedPreferences(context)
                 .edit()
                 .clear()
-                .apply();
+                .commit();
 
         PreferenceManager.setDefaultValues(context, R.xml.preferences, true);
-        mFailedResetActions.remove(mFailedResetActions.indexOf(ResetAction.RESET_PREFERENCES));
+
+        boolean clearedAdminPreferences = context
+                .getSharedPreferences(AdminPreferencesActivity.ADMIN_PREFERENCES, 0)
+                .edit()
+                .clear()
+                .commit();
+
+        boolean deletedSettingsDir = !new File(Collect.SETTINGS).exists() ||
+                deleteRecursive(new File(Collect.SETTINGS));
```

Comment:
Why is this `deleteRecursive` instead of `deleteFolderContents`? Unless there is a good reason, it should be consistent with the other folder deletes.